pytools/obj_array.py

```diff
@@ -273,15 +273,21 @@ def __matmul__(
                     self: ObjectArray2D[T],
                     other: ObjectArray2D[T],
                 /) -> ObjectArray2D[T]: ...
-
+        @overload
+        def __matmul__(
+                    self: ObjectArray2D[T],
+                    other: np.ndarray[tuple[int, int], np.dtype[Any]],
+                /) -> ObjectArray2D[T]: ...
         @overload
         def __matmul__(
                     self: ObjectArrayND[T],
                     other: ObjectArrayND[T],
                 /) -> ObjectArrayND[T] | T: ...
 
         @property
-        def flat(self) -> ObjectArray1D[T]: ...
+        def flat(self) -> Iterator[T]: ...
```

Comment:
`ndarray.flat` actually returns a `np.flatiter`, but that's generic on something with a bound on `NDArray[Any]`, so we can't really use it here. This should still be good enough?